static/src/javascripts/es6/projects/common/modules/crosswords/main.js

```diff
@@ -5,32 +5,46 @@ import React from 'react';
 
 import $ from 'common/utils/$';
 import _ from 'common/utils/_';
+import detect from 'common/utils/detect';
+import scroller from 'common/utils/scroller';
 
 import Clues from './clues';
 import Controls from './controls';
-import FocussedClue from './focussed-clue.jsx!';
+import HiddenInput from './hidden-input';
 import Grid from './grid';
 import helpers from './helpers';
 import keycodes from './keycodes';
 import persistence from './persistence';
 import loadFont from './font';
 
-// make react available to dev tool
-if (!window.React) {
-    window.React = React;
-}
+class Crossword extends React.Component {
+
+    constructor (props) {
+        super(props);
 
-const Crossword = React.createClass({
-    getInitialState () {
         const dimensions = this.props.data.dimensions;
 
         this.columns = dimensions.cols;
         this.rows = dimensions.rows;
         this.clueMap = helpers.buildClueMap(this.props.data.entries);
 
+        this.onCheat = this.onCheat.bind(this);
```

Comment:
agreed - was going to have a chat more generally about we do this, since will be common. esp wrt https://github.com/andreypopp/autobind-decorator.